src/types.ts

```diff
@@ -24,6 +24,11 @@ export interface InitConfig {
   poolName?: string
   logger?: Logger
   indy: typeof Indy
+  openMediation?: Boolean
+  mediatorInvitation?: ConnectionInvitationMessage
+  //mediatorConnectionsInvite?: ConnectionInvitationMessage
+  defaultMediatorId?: string
+  clearDefaultMediator?: boolean
```

Comment:
@dbluhm has outlined mediation flow in [this hackmd](https://hackmd.io/@dbluhm/mediated-connection-activity). For a recipient to request mediation they must establish a connection. the intended use of mediatorInvitation is to automate connection and mediation request creation and setting default mediator tag. **this code is not implemented yet**, I imagine it to be coded in `provision` which is called in the constructor of RecipientService. ACA-Py flag description is _"Connect to mediator through provided invitation, send mediation request and set as default mediator."_ which is the functionality I am voting for.

Reply:
Makes sense, I agree 👍

Reply:
I like this simplified, automatic way of mediation flow to replace current provision method in order to make it compatible with ACA-Py (or any other 'open' mediator), deprecating `mediatorUrl` init config option.

So what would be the meaning of defaultMediatorId config option? Is something like the `mediatorRecordId` from my branch? I mean: when set, agent will pick its default mediator record from wallet by querying by this id. If that's the case I don't know how can it coexist (in an elegant way) with the concept you introduced regarding a wallet mediation record containing the tag 'defaultMediator'.

Reply:
`defaultMediatorId` config option provides a way to set what mediator to be used by default at startup. This is still dependant on having an established connection with mediation granted but allows setting the `defaultMediatorId` at the start without creating a completely new connection from an invitation. The config also has clear `defaultMediatorId`. since we are using a record tag to indicate default mediator id, clear will set all record tags to a string containing false, `defaultMediatorId` will set a single record tag string to "true". So the `defaultMediatorId` config exposes set default mediator interface at the config level. I think it can coexist well as long as we have the ability to update tags on a wallet.